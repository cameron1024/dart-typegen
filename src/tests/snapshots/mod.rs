// #[test]
// fn test_foo() {
//     let dart = include_str!("./foo.dart");
//     let toml = include_str!("./foo.toml");
//
// }
