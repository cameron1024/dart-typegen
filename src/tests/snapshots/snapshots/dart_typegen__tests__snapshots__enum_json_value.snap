---
source: src/tests/snapshots/mod.rs
expression: dart
---
import "package:equatable/equatable.dart";

enum Color {
  red,
  green,
  blue;

  factory Color.fromJson(dynamic json) => switch (json) {
    "something-else" => Color.red,
    123 => Color.green,
    true => Color.blue,
    final other => throw ArgumentError("Unknown variant: $other"),
  };

  dynamic toJson() => switch (self) {
    Color.red => "something-else",
    Color.green => 123,
    Color.blue => true,
  };
}
